manifests/install.pp

```diff
@@ -10,50 +10,71 @@
 
   ensure_packages(['python-software-properties', 'software-properties-common'])
 
-  if versioncmp($::solr::version, '5.0') < 0 {
-    case $::lsbdistcodename {
+  if ! defined(Package['oracle-java8-set-default']) {
```

Comment:
Isn't this backwards ? 
If the package(oracle-java8-set-default) is not defined, then add the package, but pruge it 
Ditto oracle-java8-instller et all ?

Reply:
It's to prevent package redeclaration. Both `oracle-java8-set-default` and `oracle-java8-installer` are installed by the Oracle Java 8 PPA.